src/PointwiseFunctions/AnalyticData/CurvedWaveEquation/PureSphericalHarmonic.cpp

```diff
@@ -0,0 +1,74 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticData/CurvedWaveEquation/PureSphericalHarmonic.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/SwshInterpolation.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace CurvedScalarWave::AnalyticData {
+
+PureSphericalHarmonic::PureSphericalHarmonic(
+    double radius, const double width, const size_t l, const int m,
+    const Options::Context& /*context*/)
+    : radius_(radius), width_sq_(width * width), l_mode_(l), m_mode_(m) {
+  // LCOV_EXCL_START
+  if (abs(m_mode_) > static_cast<int>(l_mode_)) {
+    ERROR(
```

Comment:
report these errors as `PARSE_ERROR`. They throw, so you can quite easily add a few `CHECK_THROWS_WITH` tests and delete the `LCOV_EXCL_*` directives.